### DISCUSSION TITLE
I need help with PID.

### DISCUSSION THREAD

**Discussion by Yet-another-NPC-guy** [lihenggui/blocker#816] [Q&A]
I'm facing problems to identify sub processes PIDs.

su -c 'pidof (package name)/.(full sub processes name)' 

always return a different PID on every execution but:

su -c 'pidof (package name)' 

only returns one specific pid that I can comprobe it's the correct pid.

It's strange because new PID means sub processes restarts but I'm getting different PIDs and I can see how the specific sub processes still on execution.

**Comment by lihenggui** [comment]
Implementation of the pidof tool: https://cs.android.com/android/platform/superproject/+/main:external/toybox/lib/lib.c;l=1128
